pyOCD/coresight/ap.py

```diff
@@ -132,6 +132,10 @@ def writeMemory(self, addr, data, transfer_size=32):
         try:
             self.write_reg(AP_REG['TAR'], addr)
             self.write_reg(AP_REG['DRW'], data)
+        except DAPAccess.TransferFaultError as error:
+            # Annotate error with target address.
+            self._handle_error(error, num)
+            raise DAPAccess.TransferFaultError(addr)
```

Comment:
I actually started off that way, but for some reason got an exception when trying to modify the fault_address property (with appropriate setter in place) on the exception. Instead of fixing it the right way, I did this. Hadn't thought about the call stack trace—I'll do it right this time. :)